Paco/src/com/pacoapp/paco/js/bridge/JavascriptEventLoader.java

```
@@ -56,18 +56,95 @@ public String loadAllEvents() {
+  public String getEventsByQuery(String criteriaQuery) {
+	  List<Event> events = null;
+	  try{
+    	String criteriaColumns = null;
+    	String[] projectionColumns =null;
+    	String groupBy = null;
+    	String[] criteriaValue = null;
+     	String sortOrder = null;
+    	String limitRecords = null;
+    	String having = null;
+    	JSONObject criteriaQueryObj = new JSONObject(criteriaQuery);
+    	
+    	if (criteriaQueryObj.has("select")){
+    		JSONArray selectAr = criteriaQueryObj.getJSONArray("select");
+    		if(selectAr!=null){
+	    		projectionColumns = new String[selectAr.length()];
+	    		for(int j=0; j<selectAr.length();j++){
+	    			projectionColumns[j]=selectAr.getString(j);
+	    		}
+    		}
+    	}
+    	
+    	if(criteriaQueryObj.has("query")){
+    		JSONObject queryCriteria = criteriaQueryObj.getJSONObject("query");
+    		if(queryCriteria!=null){
+		    	if (queryCriteria.has("criteria")){
+		    		criteriaColumns = queryCriteria.getString("criteria");
+		    	}
+		    	
+		    	if (queryCriteria.has("values")){
+		    		JSONArray cv = queryCriteria.getJSONArray("values");
+		    		criteriaValue = new String[cv.length()];
+		    		for(int i=0; i<cv.length();i++){
+		    			criteriaValue[i]=cv.getString(i);
+		    		}
+		    	}
+    		}
+    	}
+    	
+    	if (criteriaQueryObj.has("order")){    	
+    		sortOrder = criteriaQueryObj.getString("order");
+    	}
+    	
+    	if (criteriaQueryObj.has("limit")){
+    		limitRecords = criteriaQueryObj.getString("limit");
+    	}
+    	
+    	if (criteriaQueryObj.has("group")){
+    		groupBy = criteriaQueryObj.getString("group");
+    	} 
+    	if (criteriaQueryObj.has("having")){
+    		groupBy = criteriaQueryObj.getString("having");
+    	} 
+    	events = experimentProviderUtil.findEventsByCriteriaQuery(projectionColumns,  criteriaColumns, criteriaValue, sortOrder, limitRecords, groupBy, having);
+    	
+    }catch(JSONException je){
+    	Log.error("Invalid JSON in criteria query" + je);
```

The json exception should return an error to the js call instead of jsonfiying the events (which may not be correct or may be null).